### PR TITLE
adapter: Add infrastructure for system privileges

### PR DESCRIPTION
**Pull request by jkosh44**
This commit adds default privileges to the system catalog tables and
the in memory catalog.

Currently, there's no way for a user to modify the system privileges,
it's only pre-populated with some default system privileges. In a later
commit we will add the ability to alter system privileges via the
`GRANT` command.

Works towards resolving #19997

### Motivation
This PR adds a known-desirable feature.

### Tips for Reviewer
* This PR builds off of https://github.com/MaterializeInc/materialize/pull/20005, only the most recent commit is new.

### Checklist

- [X] This PR has adequate test coverage / QA involvement has been duly considered.
- [X] This PR has an associated up-to-date [design doc](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/design/README.md), is a design doc ([template](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/design/00000000_template.md)), or is sufficiently small to not require a design.
  
- [X] If this PR evolves [an existing `$T ⇔ Proto$T` mapping](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/command-and-response-binary-encoding.md) (possibly in a backwards-incompatible way), then it is tagged with a `T-proto` label.
- [X] If this PR will require changes to cloud orchestration, there is a companion cloud PR to account for those changes that is tagged with the release-blocker label ([example](https://github.com/MaterializeInc/cloud/pull/5021)).
  
- [X] This PR includes the following [user-facing behavior changes](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/guide-changes.md#what-changes-require-a-release-note):
  -

### CODE REVIEW

**Comment by jkosh44** [comment]
This is experiencing a weird CI failure, so I'm putting it in draft if people want to hold off on reviewing.

File: src/adapter/src/catalog/builtin.rs [hunk 1]
```diff
@@ -1973,6 +1973,13 @@ pub static MZ_DEFAULT_PRIVILEGES: Lazy<BuiltinTable> = Lazy::new(|| BuiltinTable
     is_retained_metrics_object: false,
 });
 
+pub static MZ_SYSTEM_PRIVILEGES: Lazy<BuiltinTable> = Lazy::new(|| BuiltinTable {
+    name: "mz_system_privileges",
+    schema: MZ_CATALOG_SCHEMA,
+    desc: RelationDesc::empty().with_column("privileges", ScalarType::MzAclItem.nullable(false)),
```
Comment (mjibson) [inline src/adapter/src/catalog/builtin.rs:1979]: Missing the id column? Test would have found that.

**Review by mjibson — APPROVED** [review]

File: src/adapter/src/catalog/builtin.rs (hunk 1 above)
Comment (jkosh44) [reply src/adapter/src/catalog/builtin.rs:1979]: I'm not sure that I follow, what id column? This table only has a single column of type `MzAclItem`.

Is that the issue? That we need some sort of primary key column?

I also don't think I understand the test comment either. There is a SLT test that queries this table and it succeeds.

Comment (mjibson) [reply src/adapter/src/catalog/builtin.rs:1979]: My mistake. I misread the document update!

Comment (mjibson) [reply src/adapter/src/catalog/builtin.rs:1979]: Er, I mean I misread how pack works. I saw it has `id` and `row` fields and thought you were missing the id field in the desc.

**Comment by jkosh44** [comment]
After some investigation from @danhhz, this seems to be a persist related bug.

**Pull request merged**